src/packet/sym_encrypted_protected_data.rs

```diff
@@ -67,12 +66,103 @@ impl SymEncryptedProtectedData {
+        let ikm = session_key;
+
+        // Generate new salt for this seipd packet.
+        let mut salt = [0u8; 32];
+        rng.fill(&mut salt[..]);
+
+        let info = [
+            Tag::SymEncryptedProtectedData.encode(), // packet type
+            0x02,                                    // version
+            sym_alg.into(),
+            aead.into(),
+            chunk_size,
+        ];
+
+        let chunk_size_expanded = expand_chunk_size(chunk_size);
+        let hk = hkdf::Hkdf::<Sha256>::new(Some(&salt[..]), ikm);
+        let mut okm = [0u8; 42];
+        hk.expand(&info, &mut okm).expect("42");
+
+        let message_key = &okm[..sym_alg.key_size()];
+        let raw_iv_len = aead.nonce_size() - 8;
+        let iv = &okm[sym_alg.key_size()..sym_alg.key_size() + raw_iv_len];
+        let mut nonce = vec![0u8; aead.nonce_size()];
+        nonce[..raw_iv_len].copy_from_slice(iv);
+
+        let mut out = Vec::new();
+
+        let mut chunk_index: u64 = 0;
+        for chunk in plaintext.chunks(usize::try_from(chunk_size_expanded)?) {
+            let pos = out.len();
+
+            // append this next unencrypted chunk to `out`, and encrypt it in place
+            out.extend_from_slice(chunk);
+
+            let encrypt_chunk = &mut out[pos..];
+
+            let auth_tag =
+                aead.encrypt_in_place(&sym_alg, message_key, &nonce, &info, encrypt_chunk)?;
+
+            out.extend_from_slice(&auth_tag);
+
+            // Update nonce to include the next chunk index
+            chunk_index += 1;
+            let l = nonce.len() - 8;
+            nonce[l..].copy_from_slice(&chunk_index.to_be_bytes());
+        }
+
+        // Make and append final auth tag
+
+        // Associated data is extended with number of plaintext octets.
+        let size = plaintext.len() as u64;
+        let mut final_info = info.to_vec();
+        final_info.extend_from_slice(&size.to_be_bytes());
+
+        let final_auth_tag = aead.encrypt_in_place(
+            &sym_alg,
+            message_key,
+            &nonce,
+            &final_info,
+            &mut [][..], // encrypts empty string
+        )?;
+        out.extend_from_slice(&final_auth_tag);
+
+        Ok(SymEncryptedProtectedData {
+            packet_version: Default::default(),
+            data: Data::V2 {
+                sym_alg,
+                aead,
+                chunk_size,
+                salt,
+                data: out,
+            },
+        })
+    }
+
+    /// Same as [`encrypt_with_rng_seipdv1`], but uses [`thread_rng`] for RNG.
     ///
-    /// [`encrypt_with_rng`]: SymEncryptedProtectedData::encrypt_with_rng
+    /// [`encrypt_with_rng_seipdv1`]: SymEncryptedProtectedData::encrypt_with_rng_seipdv1
     /// [`thread_rng`]: rand::thread_rng
-    pub fn encrypt(alg: SymmetricKeyAlgorithm, key: &[u8], plaintext: &[u8]) -> Result<Self> {
-        Self::encrypt_with_rng(&mut thread_rng(), alg, key, plaintext)
+    pub fn encrypt_seipdv1(
```

Comment:
while we are renaming these, lets drop the convenience methods, and do only the ones that take in an rng

Reply:
I've dropped all convenience methods that produce an implicit Rng, and also dropped all `with_rng` fragments from function names (934257b9e05a55fbce639612ed2b94928f74f07a).

(I think this aligns all function names with what we've done for v6 Signatures in the previous PR.)